lib/msal-common/src/request/RequestValidator.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ScopeSet } from "./ScopeSet";
+import { StringUtils } from "./../utils/StringUtils";
+import { ClientConfigurationError } from "./../error/ClientConfigurationError";
+import { PromptValue, CodeChallengeMethodValues } from "./../utils/Constants";
+
+/**
+ * Validates server consumable params from the "request" objects
+ */
+export class RequestValidator {
+
+    /**
+     * Utility to validate scopes passed by the user in the request
+     */
+    static validateAndGenerateScopes(scopes: Array<string>, clientId: string): Array<string> {
+        // Set scopes and append default scopes
+        const scopeSet = new ScopeSet(scopes || [], clientId, true, true);
+        return scopeSet.asArray();
+    }
+
+    /**
+     * Utility to check if the `redirectUri` in the request is a non-null value
+     * @param redirectUri
+     */
+    static validateRedirectUri(redirectUri: string) : void {
+        // check if the redirectUri is null, set to default
+        if (StringUtils.isEmpty(redirectUri)) {
+            throw ClientConfigurationError.createUrlEmptyError();
```

Comment:
I would argue this function isn't need either, but if it is being used we should have a better error message thrown here.

Reply:
Updated the error message.